lightning/src/onion_message/packet.rs

```diff
@@ -136,6 +136,19 @@ impl<T: CustomOnionMessageContents> OnionMessageContents<T> {
 	}
 }
 
+impl<T: CustomOnionMessageContents> fmt::Display for OnionMessageContents<T> {
```

Comment:
Rather than defining `Display`, I think we should stick with the derived `Debug` implementation and use that. We may need to add a `Debug` type bound to `T`, but then we'll get the actual custom message instead of only the TLV type number.